lib/network/net.go

```diff
@@ -159,24 +176,47 @@ func (c *TcpConn) Send(ctx context.Context, obj ProtocolMessage) error {
 	if err != nil {
 		return fmt.Errorf("Error converting packet: %v\n", err)
 	}
+	dbg.Lvl5("Message SEND =>", fmt.Sprintf("%+v", am))
 	var b []byte
 	b, err = am.MarshalBinary()
 	if err != nil {
 		return fmt.Errorf("Error marshaling  message: %s", err.Error())
 	}
+	//c.Conn.SetWriteDeadline(time.Now().Add(timeOut))
 	// First write the size
-	var buffer bytes.Buffer
-	size := Size(len(b))
-	if err := binary.Write(&buffer, globalOrder, size); err != nil {
+	packetSize := Size(len(b))
+	if err := binary.Write(c.Conn, globalOrder, packetSize); err != nil {
 		return err
 	}
 	// Then send everything through the connection
-	buffer.Write(b)
-	c.Conn.SetWriteDeadline(time.Now().Add(timeOut))
```

Comment:
We could but then we need to make tests with that functionality, and I'm afraid when running multiple hosts on one server that the timeout fails before ? or with big messages that it fails before ? 
What do you think ?